docs/source/troubleshooting/tracing.rst

```diff
@@ -0,0 +1,106 @@
+
+.. note::
+   Files under `/sys/kernel/debug/tracing` are text files, use appropriate LabVIEW primitives when accessing them.
+
+.. note::
+   Tracing is a privileged operation. On older versions of NILRT, file ownership and/or permissions have to be adjusted when used from LabVIEW; e.g., with `chown -R lvuser:ni /sys/kernel/debug/tracing`. This change will not persist across a reboot.
+
+#. Stop any ongoing tracing and clear the buffer
+
+   .. code:: text
+
+      echo 0 > /sys/kernel/debug/tracing/tracing_on
+      echo 0 > /sys/kernel/debug/tracing/trace
+
+#. Enable tracing for all events
+
+   .. code:: text
+
+      echo 1 > /sys/kernel/debug/tracing/events/enable
+
+#. Start tracing
+
+   .. code:: text
+
+      echo 1 > /sys/kernel/debug/tracing/tracing_on
+
+#. (optional) Add custom markers to the trace
+
+   Custom markers can be added into the trace to track the application progress by writing into the trace_marker file:
+
+   .. code:: text
+
+      echo "something happened" > /sys/kernel/debug/tracing/trace_marker
+
+   Use this with caution and sparingly. Writing to the trace buffer adds additional latency.
+
+Analyzing the results
+=====================
+
+The resulting trace can be extracted by reading the `/sys/kernel/debug/tracing/trace` file directly. This will output a human readable textual representation. However this is usually pretty slow especially on large traces.
+
+A better alternative is to use the `trace-cmd` utility which can be installed from package feeds with
+
+.. code:: text
+
+   opkg install trace-cmd
+
+To extract the trace, use
+
+.. code:: text
+
+   trace-cmd extract
+
+The resulting trace file `trace.dat` can then be loaded in the front-end GUI tool `kernelshark` for off-target analysis.
+
+   .. image:: resources/trace-loaded-in-kernelshark.png
+
+`kernelshark` can be used on a Linux desktop distribution or on Windows with Windows Subsystem for Linux (WSL).
+
+LabVIEW Threads
+---------------
+
+The trace will contain threads created by LabVIEW Real-Time process (lvrt).
+The full list of these threads can be obtained by running
+
+.. code:: text
+
+   cat /proc/`pidof lvrt`/task/*/status | grep Name
+
+Some common threads are
+
+.. csv-table:: LabVIEW Real-Time threads
+   :file: resources/lvrt-threads.csv
+   :widths: 20, 60
+   :header-rows: 1
+
+Additional information and resources
+====================================
+
+* Linux Tracing Technologies - https://www.kernel.org/doc/html/latest/trace/index.html
+* Debugging the kernel using Ftrace - part 1: https://lwn.net/Articles/365835/
+* Debugging the kernel using Ftrace - part 2: https://lwn.net/Articles/366796/
+* trace-cmd: A front-end for Ftrace: https://lwn.net/Articles/410200/
+* KernelShark documentation - https://kernelshark.org/Documentation.html
+* Using KernelShark to analyze the real-time scheduler: https://lwn.net/Articles/425583/
```

Comment:
Can you make these RST links?

Reply:
Do you mean something like https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages where the link is not displayed separately but embedded inside the target?

Reply:
Yes.

Reply:
Done